onmt/model_builder.py

```diff
@@ -222,8 +216,6 @@ def fix_key(s):
     model.generator = generator
     model.to(device)
     if model_opt.model_dtype == 'fp16':
-        logger.warning('FP16 is experimental, the generated checkpoints may '
-                       'be incompatible with a future version')
         model.half()
```

Comment:
This warning was left here on purpose because we will probably break FP16 checkpoints in the future when Apex will modify its API.

Reply:
It's here now:
https://github.com/OpenNMT/OpenNMT-py/pull/1287/files?utf8=%E2%9C%93&diff=unified#diff-0c8dd0b52c7024092dfb7e6f825cebd9R66